pkg/mining/cpuMiner.go

```diff
@@ -26,18 +28,23 @@ func (c *CpuMiner) StepSize() uint64 {
 }
 
 func (c *CpuMiner) Name() string {
-	return fmt.Sprintf("CPU %d", *c)
+	return fmt.Sprintf("CPU %d", c)
 }
 
-func (c *CpuMiner) CheckRange(hash *HashSettings, start uint64, n uint64) (string, uint64, error) {
+func (c *CpuMiner) CheckRange(deadlineCtx context.Context, hash *HashSettings, start uint64, n uint64) (string, uint64, error) {
 	baseLen := len(hash.prefix)
 	hashInput := make([]byte, len(hash.prefix))
 	copy(hashInput, hash.prefix)
 
 	x := new(big.Int)
 	compareZero := big.NewInt(0)
-
 	for i := start; i < (start + n); i++ {
+		select {
+		case <-deadlineCtx.Done():
+			// Return any solution when context deadline exceeded.
+			return "", n, nil
```

Comment:
Lets add a bit more description and maybe even name the contract accordingly - `min15SinceLastSubmit` or something similiar to indicated what is this context for.
Also, I think "" string here means no solution. 
Can you please double-check as I am also not sure. 
```suggestion
			// Return any solution when context deadline exceeded.
			// In the oracle any solution will work 15min after a new challenge has been emitted  and
			// this context is canceled exactly then.
			return "any nonce will work", n, nil
```

Reply:
Yeah, that's right!